plaso/parsers/setupapi.py

```diff
@@ -145,47 +182,78 @@ def ParseRecord(self, parser_mediator, key, structure):
     Raises:
       ParseError: when the structure type is unknown.
     """
-    if key != 'logline':
-      raise errors.ParseError(
-          'Unable to parse record, unknown structure: {0:s}'.format(key))
+    if key == 'ignorable_line':
```

Comment:
what if self._last_end_time contains a value from the previous file it parsed?

Reply:
Added code to reset these values to VerifyStructure.